src/ffi/code.rs

```diff
@@ -1,156 +1,2 @@
-pub const CO_FUTURE_WITH_STATEMENT: c_int = 0x8000;
-pub const CO_FUTURE_PRINT_FUNCTION: c_int = 0x1_0000;
-pub const CO_FUTURE_UNICODE_LITERALS: c_int = 0x2_0000;
-pub const CO_FUTURE_BARRY_AS_BDFL: c_int = 0x4_0000;
-pub const CO_FUTURE_GENERATOR_STOP: c_int = 0x8_0000;
-
-pub const CO_MAXBLOCKS: usize = 20;
-pub type FreeFunc = extern "C" fn(*mut c_void) -> c_void;
-
-#[cfg_attr(windows, link(name = "pythonXY"))]
-extern "C" {
-    pub static mut PyCode_Type: PyTypeObject;
-}
-
-extern "C" {
-    pub fn _PyCode_GetExtra(
-        code: *mut PyObject,
-        index: Py_ssize_t,
-        extra: *const *mut c_void,
-    ) -> c_int;
-    pub fn _PyCode_SetExtra(code: *mut PyObject, index: Py_ssize_t, extra: *mut c_void) -> c_int;
-
-    #[cfg_attr(PyPy, link_name = "PyPyCode_New")]
-    pub fn PyCode_New(
-        argcount: c_int,
-        kwonlyargcount: c_int,
-        nlocals: c_int,
-        stacksize: c_int,
-        flags: c_int,
-        code: *mut PyObject,
-        consts: *mut PyObject,
-        names: *mut PyObject,
-        varnames: *mut PyObject,
-        freevars: *mut PyObject,
-        cellvars: *mut PyObject,
-        filename: *mut PyObject,
-        name: *mut PyObject,
-        firstlineno: c_int,
-        lnotab: *mut PyObject,
-    ) -> *mut PyCodeObject;
-    #[cfg(Py_3_8)]
-    pub fn PyCode_NewWithPosOnlyArgs(
-        argcount: c_int,
-        posonlyargcount: c_int,
-        kwonlyargcount: c_int,
-        nlocals: c_int,
-        stacksize: c_int,
-        flags: c_int,
-        code: *mut PyObject,
-        consts: *mut PyObject,
-        names: *mut PyObject,
-        varnames: *mut PyObject,
-        freevars: *mut PyObject,
-        cellvars: *mut PyObject,
-        filename: *mut PyObject,
-        name: *mut PyObject,
-        firstlineno: c_int,
-        lnotab: *mut PyObject,
-    ) -> *mut PyCodeObject;
-    #[cfg_attr(PyPy, link_name = "PyPyCode_NewEmpty")]
-    pub fn PyCode_NewEmpty(
-        filename: *const c_char,
-        funcname: *const c_char,
-        firstlineno: c_int,
-    ) -> *mut PyCodeObject;
-    pub fn PyCode_Addr2Line(arg1: *mut PyCodeObject, arg2: c_int) -> c_int;
-    pub fn PyCode_Optimize(
-        code: *mut PyObject,
-        consts: *mut PyObject,
-        names: *mut PyObject,
-        lnotab: *mut PyObject,
-    ) -> *mut PyObject;
-
-    #[cfg(PyPy)]
-    #[link_name = "PyPyCode_Check"]
-    pub fn PyCode_Check(op: *mut PyObject) -> c_int;
-
-    #[cfg(PyPy)]
-    #[link_name = "PyPyCode_GetNumFree"]
-    pub fn PyCode_GetNumFree(op: *mut PyCodeObject) -> Py_ssize_t;
-}
-
-#[inline]
-#[cfg(not(PyPy))]
-pub unsafe fn PyCode_Check(op: *mut PyObject) -> c_int {
-    (Py_TYPE(op) == &mut PyCode_Type) as c_int
-}
-
-#[inline]
-#[cfg(not(PyPy))]
-pub unsafe fn PyCode_GetNumFree(op: *mut PyCodeObject) -> Py_ssize_t {
-    crate::ffi::tupleobject::PyTuple_GET_SIZE((*op).co_freevars)
-}
+// TODO: newtype
+pub use crate::ffi::PyCodeObject;
```

Comment:
The definition in `Include/code.h` is `typedef struct PyCodeObject PyCodeObject`, where the actual def is in `Include/cpython/code.h`.

I plan to change this into a newtype, but this requires some other code changes I'm not prepared to make yet. Just making a note here.

Reply:
I think what we need is 
```rust
#[cfg(Py_LIMITED_API)]
opaque_struct!(PyCodeObject);
```
Also, we have to remove `#[cfg(not(Py_LIMITED_API)]` from `mod code;`.

Reply:
I've made this change

Reply:
Also, let's remove `pub use crate::ffi::PyCodeObject;`.
Since we import all items in `mod.rs`, this is not so mearningful.